tests/tracer/test_propagation.py

```diff
@@ -20,19 +20,223 @@
 from ddtrace.propagation.http import _HTTP_HEADER_B3_SINGLE
```

Comment:
we should probably have a test case that creates a trace with multiple spans. It looks like by this implementation that the propagated tags will be applied to each span but this isn't enforced or confirmed by any tests. Maybe a snapshot test would be useful for ensuring this?